### DISCUSSION TITLE
Relative value animation track.

### DISCUSSION THREAD

**Discussion by colorworlds** [godotengine/godot-proposals#4057] [Engine Core]
Now the value of the animation track is fixed.

Sometimes this lacks flexibility.  
For example, if a `Sprite Node` already has a position animation, it will be troublesome to change the initial position of the `Sprite`.

For example, There is a `Sprite` has a track to change its position.  
The animation moves the sprite from position Vector2(0,100) to position Vector2(0,200) in 3 seconds.  
Similar to the following:

```
position track:Vector2(0,100),  ,  , Vector2(0,200)
timeline         :0          ,1 ,2 , 3
```

If I move the initial position of the `Sprite` to Vector2(200,200).  
I may need to change the position animation to move the `Sprite` from position Vector2(200 + 0, 200 + 100) to position Vector2(200 + 0,  200 + 200) in 3 seconds.    
At present, I must manually modify each position of the `Sprite` on the animation track.   
The track may change to following:

```
position track:Vector2(200+0,200+100),  ,  , Vector2(200+0,200+200)
timeline      :0                 , 1 ,2 , 3
```

This is what I mean by "Relative value animation track".   
At this time, the value in the track represents not the current value of the Sprite, but the relative value relative to the initial value of the Sprite.  

Using Tween Node can do similar things, but it must use code.

**Comment by Mickeon** [comment]
A pretty nice [mock-up for a reworked Animation Editor has been proposed](https://github.com/godotengine/godot-proposals/issues/3950) which I recall has this.

**Reply by colorworlds** [reply]
Cool. Someone has done a lot of work

**Comment by theraot** [comment]
The current approach would be parenting the node. You can move the parent of the Sprite from your code or in the editor, and let the animation move the Sprite itself inside the parent, which is relative to the parent.
